File: 02_csharp_module/08_interfaces/LongDeposit.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Interfaces
{  
    class LongDeposit : Deposit, IProlongable 
    {
        public LongDeposit(decimal amount, int period) : base(amount, period)
        {
            this.amount = amount;
            this.period = period;
        }
       
        public bool CanToProlong() 
        {
            return period <= 36;            
        }
        public override decimal Income()
        {
            double interest = 0.15;
            double balanceEnd = 0;
            double amountD = System.Convert.ToDouble(amount);

            if (period <= 6)
                return 0;
            else
            {
                for (double i = 6; i <= period; i++)
                {

                    balanceEnd = amountD * System.Math.Pow(1 + interest, i - 6);
                }

                decimal balanceEndD = System.Convert.ToDecimal(balanceEnd);
                decimal difference = balanceEndD - amount;
                return difference;
            }
        }
    }
}

File: 03_java_module/06/LongDeposit.java
package com.epam.test.automation.java.practice7;

import java.math.BigDecimal;

/**
 * <summary>
 * Implement class according to description of task.
 * </summary>
 */
class LongDeposit extends Deposit
        {
public LongDeposit(BigDecimal amount, int period)
        {
            super(amount, period);
        this.amount = amount;
        this.period = period;

        }
@Override
public  BigDecimal income()
        {
        double interest = 0.15;
        BigDecimal balanceEnd = BigDecimal.ZERO;
        BigDecimal amountD = amount;

        if (period <= 6)
            return BigDecimal.ZERO;
        else
        {
            for (double i = 6; i <= period; i++)
            {
                balanceEnd = amountD.add(BigDecimal.valueOf(Math.pow(1 + interest, i - 6)));
            }

        return balanceEnd.subtract(amount);
        }
        }
        }
